api/v1alpha1/workloadconfig.go

```diff
@@ -0,0 +1,110 @@
+package v1alpha1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+func init() {
+	SchemeBuilder.Register(&WorkloadConfiguration{})
+	SchemeBuilder.Register(&WorkloadConfigurationList{})
+}
+
+// WorkloadConfiguration allows additional tuning of control plane workloads
+// +kubebuilder:resource:path=workloadconfigurations,shortName=wcfg;wcfgs,scope=Namespaced
+// +kubebuilder:storageversion
+// +kubebuilder:subresource:status
+// +kubebuilder:object:root=true
+type WorkloadConfiguration struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec contains the workload configuration spec
+	// +optional
+	Spec WorkloadConfigurationSpec `json:"spec,omitempty"`
+
+	// Status contains the workload configuration status
+	// +optional
+	Status WorkloadConfigurationStatus `json:"status,omitempty"`
+}
+
+type WorkloadConfigurationStatus struct {
+}
+
+type WorkloadConfigurationSpec struct {
+	// Labels that should be applied to all deployments in the control plane
+	// +optional
+	Labels map[string]string `json:"labels,omitempty"`
+
+	// Annotations that should be applied to all deployments in the control plane
+	// +optional
+	Annotations map[string]string `json:"annotations,omitempty"`
+
+	// Deployments contains configuration details for control plane component deployments
+	// +listType=map
+	// +listMapKey=name
+	// +optional
+	Deployments []DeploymentWorkloadConfiguration `json:"deployments,omitempty"`
```

Comment:
How will people know what the names of the deployments are without looking through the code?